packages/@o3r/design/src/core/design-token/parsers/design-token.parser.ts

```diff
@@ -30,11 +31,36 @@ const getExtensions = (nodes: NodeReference[], context: DesignTokenContext | und
   }, {} as DesignTokenGroupExtensions & DesignTokenExtensions);
 };
 const getReferences = (cssRawValue: string) => Array.from(cssRawValue.matchAll(tokenReferenceRegExp)).map(([,tokenRef]) => tokenRef);
+const applyConversion = (token: DesignTokenVariableStructure, value: string) => {
+  if (typeof token.extensions.o3rUnit === 'undefined' || typeof token.extensions.o3rRatio === 'undefined') {
+    return value;
+  }
+
+  const splitValue = splitValueNumericRegExp.exec(value);
+  if (!splitValue) {
+    return value;
+  }
+
+  const [, floatValue, unit] = splitValue;
+
+  const newValue = value.replace(floatValue, (parseFloat((parseFloat(floatValue) * token.extensions.o3rRatio).toFixed(3))).toString());
```

Comment:
I think the 'o3rRatio' can be undefined at this point in the code. maybe you could do a check previously

Reply:
if `token.extensions.o3rRatio` is undefined it will have returned at line 36 already

Reply:
indeed :) I missed it